iris-mpc/src/services/processors/batch.rs

```diff
@@ -137,13 +139,110 @@ impl<'a> BatchProcessor<'a> {
     }
 
     async fn poll_messages(&mut self) -> Result<(), ReceiveRequestError> {
-        // let max_batch_size = self.config.max_batch_size;
+        let max_batch_size = self.config.max_batch_size;
         let queue_url = &self.config.requests_queue_url;
+        let polling_timeout =
+            tokio::time::Duration::from_secs(self.config.batch_polling_timeout_secs);
 
-        // Poll until we have enough messages
-        // temporary hack for staging to only process 1 message at a time
-        // this helps with the correctness test
-        while self.msg_counter < 1 {
+        // Try to fill the batch with a timeout
+        match tokio::time::timeout(
+            polling_timeout,
+            self.poll_until_batch_size(max_batch_size, queue_url),
+        )
```

Comment:
I think this could also lead to some edge cases where the get_message call is cancelled due to this timeout, and messages are already in flight. This means that subsequent calls to get_message actually block/return nothing, since the in-flight message is not deleted and we have to wait for its visibility timeout to elapse before the system is ready again.
We encountered this in document-mpc